modules/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,215 @@
+GNU General Public License for more details.
+*/
+class Jetpack_Simple_Payments {
+	// These have to be under 20 chars because that is CPT limit.
+	static $post_type_order = 'jp_pay_order';
+	static $post_type_product = 'jp_pay_product';
+
+	static $shortcode = 'simple-payment';
+
+	// Classic singleton pattern:
+	private static $instance;
+	private function __construct() {}
+	static function getInstance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+		return self::$instance;
+	}
+
+	private function register_scripts() {
+		/**
+		 * Paypal heavily discourages putting that script in your own server:
+		 * @see https://developer.paypal.com/docs/integration/direct/express-checkout/integration-jsv4/add-paypal-button/
+		 */
+		wp_register_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js' );
+		wp_register_script( 'paypal-express-checkout', plugins_url( '/paypal-express-checkout.js', __FILE__ ) , array( 'paypal-checkout-js' ) );
+	}
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+	}
+	private function register_shortcode() {
+		add_shortcode( self::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	public function init_hook_action() {
+		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
+		$this->register_scripts();
+		$this->register_shortcode();
+		$this->setup_cpts();
+	}
+
+	function parse_shortcode( $attrs, $content = false ) {
+		if( empty( $attrs['id'] ) ) {
+			return;
+		}
+		$product = get_post( $attrs['id'] );
+		if( is_wp_error( $product ) ) {
+			return;
+		}
+		if( $product->post_type !== self::$post_type_product ) {
+			return;
+		}
+
+		// We allow for overriding the presentation labels
+		$data = shortcode_atts( array(
+			'dom_id' => uniqid( 'jp_simple_payments__button_' . $product->ID . '_' ),
+			'class' => 'jp_simple_payments__' . $product->ID,
+			'title' => get_the_title( $product ),
+			'description' => apply_filters( 'the_content', $product->post_content ),
+			'cta' => get_post_meta( $product->ID, 'spay_cta', true ),
+			'multiple' => get_post_meta( $product->ID, 'spay_multiple', true )
+		), $attrs );
+		$data['price'] = $this->format_price(
+			get_post_meta( $product->ID, 'spay_price', true ),
+			get_post_meta( $product->ID, 'spay_currency', true ),
+			$data
+		);
+
+		wp_enqueue_script( 'paypal-express-checkout' );
+		wp_add_inline_script( 'paypal-express-checkout', "try{PaypalExpressCheckout.renderButton( {$data['dom_id']} );}catch(e){}" );
+
+		return $this->output_shortcode( $data );
+	}
+
+	function output_shortcode( $data ) {
+		$items="";
+		// TODO: tie number of items to request
+		if( $data['multiple'] ) {
+		       $items='<div class="jp_simple_payments__items" ><input type="number" value="1"></div>';
+		}
+		$output = <<<TEMPLATE
+<div class="{$data[ 'class' ]} jp_simple_payments__wrapper">
```

Comment:
I think we should avoid using underscores for selectors: https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/#selectors

Wouldn't `jetpack-` be better than just `jp-`?